tests/integration/test_models_vision_text_config_registry.py

```
@@ -0,0 +1,1675 @@
+    def test_streaming_guard_iterator_fires_on_exhaustion(self) -> None:
+        from file_organizer.models.text_model import _GuardedIterator
+
+        fired: list[bool] = []
+
+        def gen():
+            yield "x"
+
+        gi = _GuardedIterator(gen(), lambda: fired.append(True))
+        with pytest.raises(StopIteration):
+            next(gi)
+            next(gi)
```

Ruff (0.15.9)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)